### PR TITLE
[Hexagon] Float and quantized dense operators with schedules

### PR DESCRIPTION
**Pull request by joshherr-quic**
This PR implements dense operators for float types and quantized types. The quantized implementation uses floating point numbers for its intermediate compute type, fixed point will be investigated in the future.

float16 accuracy is questionable. Needs further investigation in an actual model (not just a unittest).

### CODE REVIEW

**Comment by mehrdadh** [comment]
@tvm-bot rerun

**Review by mehrdadh — CHANGES_REQUESTED** [review]
Let's wait for this PR and rerun this again. After this PR, please move the test file to slice_op sub directory.
https://github.com/apache/tvm/pull/13271

Also, to help with the pylint with test files, please add the test file to pylint script:
tests/lint/pylint.sh

**Comment by joshherr-quic** [comment]
@tvm-bot rerun

**Comment by joshherr-quic** [comment]
PR is failing for other targets, not sure what is going on.

CC: @mehrdadh

**Comment by kparzysz-quic** [comment]
Ping.  The lint issues have been addressed and the PR has been rebased.

**Comment by kparzysz-quic** [comment]
Ping @mehrdadh

**Comment by kparzysz-quic** [comment]
@mehrdadh

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+def transformed_expected_output_np(expected_output_np, layout):
+    return transform_numpy(expected_output_np, "nc", layout)
+
+
+@tvm.testing.fixture
+def transformed_input_np(input_np, layout):
+    return transform_numpy(input_np, "nc", layout)
+
+
+# TODO(joshherr-quic): transforming weight forces us to put it in vtcm. Crashes at runtime in vtcm
```
Comment (mehrdadh) [inline tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py:101]: please remove these comments.

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+@tvm.testing.fixture
+def transformed_input_quant(input_quant, layout):
+    if input_quant is None:
+        return None
+    input_quant["data"] = transform_numpy(input_quant["data"], "nc", layout)
+    return input_quant
+
+
+# @tvm.testing.fixture
```
Comment (mehrdadh) [inline tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py:115]: same here

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+# def transformed_weight_quant(weight_quant, layout):
+#     weight_quant["data"] = transform_numpy(weight_quant["data"], "nc", layout)
+#     return weight_quant
+
+# Test combinations of the following:
```
Comment (mehrdadh) [inline tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py:120]: These comments are also not needed, testing parameters show the combination

**Review by mehrdadh — CHANGES_REQUESTED** [review]
sorry for the delay, looks good.
Please address these few comments before we merge it

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py (hunk 1 above)
Comment (kparzysz-quic) [reply tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py]: Removed.

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py (hunk 2 above)
Comment (kparzysz-quic) [reply tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py]: Removed.

File: tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py (hunk 3 above)
Comment (kparzysz-quic) [reply tests/python/contrib/test_hexagon/topi/slice_op/test_dense_slice.py]: Removed.

**Review by mehrdadh — APPROVED** [review]

**Pull request merged**